### PR TITLE
Update Jenkinsfile.cd

### PR DESCRIPTION
**Pull request by tskinn**
In preparation for moving to a new branching strategy, this PR updates Jenkinsfile.cd to handle git push events for the master branch, triggering a build and deploy to dev and staging.

ref: https://github.com/department-of-veterans-affairs/va.gov-team/issues/1750

### CODE REVIEW

**Comment by omgitsbillryan** [comment]
😱 I thought GitHub had comment length limits, the bot here proved that to be false. 😱

**Comment by ElijahLynn** [comment]
> �[31m        SQLSTATE[40001]: Serialization failure: 1213 Deadlock found when trying to get lock; try restarting transaction: INSERT INTO {node_field_data} (nid, vid, type, langcode, status, title, uid, created, changed, promote, sticky, default_langcode, revision_translation_affected) VALUES (:db_insert_placeholder_0, :db_insert_placeholder_1, :db_insert_placeholder_2, :db_insert_placeholder_3, :db_insert_placeholder_4, :db_insert_placeholder_5, :db_insert_placeholder_6, :db_insert_placeholder_7, :db_insert_placeholder_8, :db_insert_placeholder_9, :db_insert_placeholder_10, :db_insert_placeholder_11, :db_insert_placeholder_12); Array

This is fixed SQL setting we need to change, `tx_isolation: READ-COMMITTED`, which we already applied to all the RDS servers a while ago, but CI uses an on-instance MySQL server. cc @jonpugh 

image

File: Jenkinsfile.cd [hunk 1]
```diff
@@ -1,42 +1,76 @@
...
-    stage('Deploy to cms[vagov-dev]') {
-      steps {
-        build job: "deploys/cms-vagov-dev", parameters: [
-          stringParam(name: 'app', value: 'cms'),
-          booleanParam(name: 'notify_slack', value: true),
-          stringParam(name: 'ref', value: commit),
-          booleanParam(name: 'migration_status', value: false)
-        ] , wait: true
+    stage('Deploy to dev') {
+      if (IS_DEV_BRANCH) {
```
Comment (ElijahLynn) [inline Jenkinsfile.cd]: As discussed on zoom, add `shouldBail()` here.

File: Jenkinsfile.cd [hunk 2]
```diff
@@ -1,42 +1,76 @@
...
+            booleanParam(name: 'notify_slack', value: true),
+            stringParam(name: 'ref', value: commit),
+            booleanParam(name: 'migration_status', value: false)
+          ] , wait: false
+        }
+      }
+    }
+
+    stage('Deploy to staging') {
+      if (IS_STAGING_BRANCH) {
```
Comment (ElijahLynn) [inline Jenkinsfile.cd]: As discussed on zoom, add `shouldBail()` here.

**Comment by ElijahLynn** [comment]
Closing/opening to destroy ENV as there is an issue with it.

**Pull request merged**